HPXMLtoOpenStudio/resources/vehicle.rb

```diff
@@ -0,0 +1,170 @@
+  # @param schedules_file [SchedulesFile] SchedulesFile wrapper class instance of detailed schedule files
+  # @return [nil]
+  def self.apply(runner, model, spaces, hpxml_bldg, schedules_file)
+    hpxml_bldg.vehicles.each do |vehicle|
+      next unless vehicle.vehicle_type == Constants::ObjectTypeBatteryElectricVehicle
+
+      apply_electric_vehicle(runner, model, spaces, hpxml_bldg, vehicle, schedules_file)
+    end
+  end
+
+  # Apply an electric vehicle to the model using the battery.rb Battery class, which assigns OpenStudio ElectricLoadCenterStorageLiIonNMCBattery, ElectricLoadCenterDistribution, OtherEquipment, and EMS objects.
+  # Custom adjustments for EV operation are made with an EMS object within this class and in the Battery class.
+  # An EMS program models the effect of ambient temperature on the effective power output.
+  # An EMS program writes a 'discharge offset' variable to omit this from aggregate home electricity outputs.
+  # If no charging/discharging schedule is provided, then the electric vehicle is not modeled.
+  # Bi-directional charging is not currently implemented
+  #
+  # @param runner [OpenStudio::Measure::OSRunner] Object typically used to display warnings
+  # @param model [OpenStudio::Model::Model] OpenStudio Model object
+  # @param spaces [Hash] Map of HPXML locations => OpenStudio Space objects
+  # @param hpxml_bldg [HPXML::Building] HPXML Building object representing an individual dwelling unit
+  # @param vehicle [HPXML::Vehicle] Object that defines a single electric vehicle
+  # @param schedules_file [SchedulesFile] SchedulesFile wrapper class instance of detailed schedule files
+  # @return [nil]
+  def self.apply_electric_vehicle(runner, model, spaces, hpxml_bldg, vehicle, schedules_file)
+    model.getElectricEquipments.sort.each do |ee|
+      if ee.endUseSubcategory.start_with? Constants::ObjectTypeMiscElectricVehicleCharging
+        runner.registerWarning('Electric vehicle was specified as a plug load and as a battery, vehicle charging will be modeled as a plug load.')
+        return
+      end
+    end
+
+    # Assign charging and vehicle space
+    ev_charger = vehicle.ev_charger
+    if ev_charger.nil?
+      runner.registerWarning('Electric vehicle specified with no charger provided; battery will not be modeled.')
+      return
+    else
+      ev_charger.additional_properties.space = Geometry.get_space_from_location(ev_charger.location, spaces)
+      vehicle.location = ev_charger.location
+      vehicle.additional_properties.space = Geometry.get_space_from_location(vehicle.location, spaces)
+    end
+
+    # Calculate average discharge power
+    hrs_driven_year = vehicle.hours_per_week / 7 * UnitConversions.convert(1, 'yr', 'day') # hrs/year
+    ev_annl_energy = vehicle.energy_efficiency * vehicle.miles_per_year # kWh/year
+    vehicle.rated_power_output = UnitConversions.convert(ev_annl_energy / hrs_driven_year, 'kw', 'w') # W
```

Comment:
This is set to the E+ in `battery.rb` as the `DesignStorageControlDischargePower`. In practice this is the _effective_ discharge power, but it is still assigned as the design power because 1) we would otherwise get an error when setting `elcd.setStorageOperationScheme('TrackChargeDischargeSchedules')` and 2) the discharge power is overwritten in the EMS program below, and the value is not actually used.

The thought is that using another power output could be confusing to a user, particularly one that is not at all related to the schedule and efficiency inputs provided.